### PR TITLE
improve error message for mismatched pipelines

### PR DESCRIPTION
**Pull request by ZhengLin-Li**
improve error message for mismatched pipelines

fix issue: #18040

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] #18040
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by ZhengLin-Li** [comment]
R: @tvalentyn

could I ask a question that why there are two checks were failed?

**Comment by tvalentyn** [comment]
Apologies for the long response, I was OOO for a bit. Thanks for the fix! 

Precommit errors are most likely a flake, we can re-run the suite. Lint errors is visible if you follow the "Details" -> Console Output link. The error is:

```
apache_beam  conftest.py  gen_protos.py  setup.py
Running pylint...
************* Module apache_beam.transforms.util
apache_beam/transforms/util.py:238:0: C0301: Line too long (88/80) (line-too-long)

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, -0.00)

```

**Comment by tvalentyn** [comment]
see also: s.apache.org/beam-python-dev-wiki which talks about how to run linter locally during development.

File: sdks/python/apache_beam/transforms/util.py [hunk 1]
```diff
@@ -234,7 +234,8 @@ def expand(self, pcolls):
     for pcoll in pcolls.values():
       self._check_pcollection(pcoll)
       if self.pipeline:
-        assert pcoll.pipeline == self.pipeline
+        assert pcoll.pipeline == self.pipeline, (
+        'mismatched pipelines, input PCollections must all belong to the same pipeline')
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/util.py:238]:
```suggestion
        'All input PCollections must belong to the same pipeline.')
```

File: sdks/python/apache_beam/transforms/util.py [hunk 2]
```diff
@@ -234,7 +234,8 @@ def expand(self, pcolls):
     for pcoll in pcolls.values():
       self._check_pcollection(pcoll)
       if self.pipeline:
-        assert pcoll.pipeline == self.pipeline
+        assert pcoll.pipeline == self.pipeline, (
+        'All input PCollections must belong to the same pipeline.')
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/util.py:238]:
```suggestion
            'All input PCollections must belong to the same pipeline.')
```

**Review by tvalentyn — APPROVED** [review]

**Comment by tvalentyn** [comment]
Linter errors should be fixed, we can merge once tests run.

**Pull request merged**